Add configuration options to support different driver modes (#372)
digitalocean/csi-digitalocean: README.md

The [DigitalOcean Kubernetes](https://www.digitalocean.com/products/kubernetes/) product comes with the CSI driver pre-installed and no further steps are required.

---
**Driver modes:**

> By default, driver runs in both controller and node mode. It would create disk Volumes on DigitalOcean infrastructure and mount them on the required node.

Reviewer: ```suggestion
By default, the driver supports both the [controller and node mode](https://kubernetes-csi.github.io/docs/deploying.html): it can manage DigitalOcean Volumes via the cloud API and mount them on the required node. The actually used mode is determined by how the driver is deployed and configured.
```

> Driver can be run in **controller only mode** outside DigitalOcean droplets.

Reviewer: ```suggestion
When running outside of DigitalOcean droplets, the driver can only function in **controller mode**. This requires to set the `--region` flag (a valid DigitalOcean region slug) in addition to the other flags. The `--region` flag should **not** be set when running the driver on DigitalOcean droplets.
```

> To use this mode `--region` flag (valid DigitalOcean region slug) must be provided together with `--token` flag (DigitalOcean API token).

Reviewer: ```suggestion

```

> Alternatively driver can be run in **node only mode** on DigitalOcean droplets. Driver would only handle node related requests like mount volume.

Reviewer: ```suggestion
When running in **node mode only** on DigitalOcean droplets, the `token` flag is not needed (and consequently, no API token secret needs to be deployed). Additionally, in this case the `--region` and `--token` flags must not be specified together.
```

Reviewer: Does this suggestion still need to be applied?

Author: I think yes, as it is possible to run driver in this mode if someone else would find a use case for it as we did.
Added additional sentence to be more clear how it's ran in this mode.

> To us this mode `--region` and `--token` flags must not be provided.

Reviewer: ```suggestion
```

> Skip secret creation (section 1. in following deployment instructions) when using **node only mode** as API token is not required.

Reviewer: Please drop the sentence as I merged it with the upper paragraph.

Reviewer: Can we? :)

Author: Yeah, node mode does not interact with DigitalOcean API, it just performs node operations: mount, unmount, etc. On the fork I'm using it without token, thus secret is not required for this mode.

---
**Driver modes:**

By default, the driver supports both the [controller and node mode.](https://kubernetes-csi.github.io/docs/deploying.html)
It can manage DigitalOcean Volumes via the cloud API and mount them on the required node.
The actually used mode is determined by how the driver is deployed and configured.
The suggested release manifests provide separate deployments for controller and node modes, respectively.

When running outside of DigitalOcean droplets, the driver can only function in **controller mode**.
This requires to set the `--region` flag to a valid DigitalOcean region slug in addition to the other flags.

The `--region` flag **must not** be set when running the driver on DigitalOcean droplets.

Alternatively driver can be run in **node only mode** on DigitalOcean droplets.
Driver would only handle node related requests like mount volume. Driver runs in **node only mode** when `--token` flag is not provided.

Skip secret creation (section 1. in following deployment instructions) when using **node only mode** as API token is not required.

> | Modes                                     |  `--token` flag  |  `--region` flag |

Reviewer: This table is super useful, thanks a bunch 👍
